controlplane/kubeadm/controllers/controller.go

```diff
@@ -384,34 +384,32 @@ func (r *KubeadmControlPlaneReconciler) reconcileDelete(ctx context.Context, clu
 	logger := r.Log.WithValues("namespace", kcp.Namespace, "kubeadmControlPlane", kcp.Name, "cluster", cluster.Name)
 	logger.Info("Reconcile KubeadmControlPlane deletion")
 
+	// Gets all machines, not just control plane machines.
+	allMachines, err := r.managementCluster.GetMachinesForCluster(ctx, util.ObjectKey(cluster))
```

Comment:
what about MachinePools?

Reply:
This PR is just shuffling things around to fix a bug, but this bit is not changing from what we have currently.
However this is a good point and I open an issue to track that we are missing machine pools here

Reply:
@CecileRobertMichon see https://github.com/kubernetes-sigs/cluster-api/issues/3880